### PR TITLE
implement muldiv operation for u128

### LINKED ISSUE #46: Add overflow-free mulDiv method for all types
Triplet of `a * b / c` integer operations may overflow. So will be great have `u128.mulDiv` operation which avoid overflow in multiplication part.

### PR DESCRIPTION
**Pull request by MaxGraey**
Partially fix #46

### CODE REVIEW

File: assembly/__tests__/u128.spec.as.ts
Comment (MaxGraey) [inline assembly/__tests__/u128.spec.as.ts]: Realized we have some tests for reminder which didn't pass

**Comment by luciotato** [comment]
Who should review this? can you merge so this moves into AssemblyScript/NEAR-as?

**Review by willemneal — APPROVED** [review]
LGTM, the tests provided seem to span the edge cases, but I am getting a few compiler warnings:

```
WARNING AS228: Function 'assembly/integer/u128/u128#preDec' is virtual and will not be inlined.

   --mask;
   ~~~~~~
 in assembly/globals.ts(282,3)

WARNING AS228: Function 'assembly/integer/u128/u128#preInc' is virtual and will not be inlined.

       ++q;
       ~~~
 in assembly/globals.ts(290,7)

✔ assembly/__tests__/safe_u128.spec.as.ts Pass: 59 / 59 Todo: 0 Time: 31.354ms
WARNING AS226: Expression resolves to unusual type 'usize'.

     expect(!changetype<u128>(null)).toBe(true);
                              ~~~~
 in assembly/__tests__/u128.spec.as.ts(307,30)

WARNING AS226: Expression resolves to unusual type 'usize'.

      !(u128.fromBytes(changetype<u8[]>(null)));
                                        ~~~~
 in assembly/__tests__/u128.spec.as.ts(1023,40)

WARNING AS226: Expression resolves to unusual type 'usize'.

       !(u128.fromBytes(changetype<u8[]>(null), true));
                                         ~~~~
 in assembly/__tests__/u128.spec.as.ts(1029,41)
```

Perhaps the warnings could be addressed in a separate PR.

**Comment by willemneal** [comment]
@MaxGraey, what is the status here?  Should I ask someone else to review it?

**Comment by MaxGraey** [comment]
It's ready except `safe.u128#muldiv` doesn't check overflow which still present for cases like 
`u128.muldiv(u128.Max, u128.Max, u128.One)`

But checks make sense do before actual muldiv operation or in between muldiv stages. I have not yet figured out how to do it simply and efficiently for safe.u128.

**Pull request merged**
